### PR TITLE
Add support for selecting which pv_estimate to use for forecasts

### PR DESCRIPTION
**Pull request by Zachoz**
Solcast provides three different forecasts for each site based around different cloud cover scenarios.

According to [their docs](https://docs.solcast.com.au/#25ff8ad7-e2a8-44be-9d2e-62e0f73cefd6), pv_estimate10 is based on a cloudier than expected scenario, and pv_estimate90 is for a less cloudy than expected scenario.

 The graph on their API toolkit page gives a better visualisation of how the estimates differ day by day:
image

I've been running with pv_estimate10 set for the last 2 months now and have founded I'm getting far more accurate predictions for where I live.

### CODE REVIEW

File: custom_components/solcast_solar/__init__.py [hunk 1]
```diff
@@ -243,6 +275,16 @@ async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
         config_entry.version = 6
 
+    #new 4.0.16
```
Comment (rany2) [inline custom_components/solcast_solar/__init__.py:278]: You can use something like:

```python3
    if config_entry.version == 6:
        new_options = {**config_entry.options, CONF_PV_ESTIMATE: "pv_estimate"}
        config_entry.version = 7
        hass.config_entries.async_update_entry(config_entry, options=new_options)
```

**Review by rany2 — CHANGES_REQUESTED** [review]

File: custom_components/solcast_solar/__init__.py (hunk 1 above)
Comment (oziee) [reply custom_components/solcast_solar/__init__.py:278]: im going a different way with this change.. adding it to the CONFIG section of the integration directly
Screenshot 2024-03-29 at 10 44 06 pm

**Comment by oziee** [comment]
added your idea @Zachoz but did it in a different way

**Comment by BJReplay** [comment]
Is there any chance of suggesting two more options for this @oziee?  I was just looking to see if I could template this up, but I realise that I can't really, because it is part of the configuration, and once set, only one set of forecasts is returned to sensors.

I was going to template up a remaining for today and next day from the estimate and esitimate10 values - because I've been finding that estimate is too high and estimate10 is too low in the cloudy weather we've been having in Melbourne.

Ideally, I'd like to be able to (particularly in Autumn and winter) choose "estimate30" - which really just is an average of estimate10 and estimate for each period / day and have that then flow through to each of the forecast entities, just as choosing estimate or estimate10 or estimate90 does now.

So the values would be:

- Estimate10 - Returns Estimate10
- Estimate30 - Returns the average of Estimate10 and Estimate
- Estimate - Returns Estimate
- Estimate70 - Returns the average of Estimate and Estimate90
- Estimate90 - Returns Estimate90

I know it is more complex, and it is beyond my ability to even work out from browsing the code base to work out how you currently select which set of forecasts to return; I have found where you store each of 10, standard and 90, but not where they are returned, so I can't even see if returning an average of two is simple or complex or impossible.

**Comment by oziee** [comment]
@BJReplay sorry no.. it would be OTT trying to work out last value plus this value or next value plus this value.. and if data is missing

**Comment by BJReplay** [comment]
No worries.  I'm lucky - I'm an early user with 50 calls per day, so I'm going to see if I can add two integrations, and template up one set to Estimate10 and one set to Estimate.

**Pull request closed without merging**